internal/integration/testcase_test.go

```diff
@@ -110,8 +128,17 @@ func (t testcase) createClusterQueueWithCleanup() *buildkite.ClusterQueue {
 			return
 		}
 
-		_, err := t.Buildkite.ClusterQueues.Delete(cfg.Org, cfg.ClusterUUID, *queue.ID)
-		if err != nil {
+		if err := roko.NewRetrier(
+			roko.WithMaxAttempts(5),
+			roko.WithStrategy(roko.Exponential(time.Second, 5*time.Second)),
+		).DoWithContext(context.Background(), func(r *roko.Retrier) error {
+			_, err := t.Buildkite.ClusterQueues.Delete(cfg.Org, cfg.ClusterUUID, *queue.ID)
+			if err != nil {
+				// There is a small chance that we are deleting queue too soon before queue realize agent has disconnected.
+				return err
+			}
+			return nil
```

Comment:
What you have here is totally fine, but I'm very tempted to suggest
```suggestion
			// There is a small chance that we are deleting queue too soon before queue realize agent has disconnected.
			_, err := t.Buildkite.ClusterQueues.Delete(cfg.Org, cfg.ClusterUUID, *queue.ID)
			return err
```